File: Downloads/champions-guild-bot/commands/serverinfo.js
const { SlashCommandBuilder, EmbedBuilder } = require('discord.js');

module.exports = {
  data: new SlashCommandBuilder()
    .setName('serverinfo')
    .setDescription('Shows stats about this server — members, bots, channels, and more.'),

  async execute(interaction) {
    await interaction.deferReply();
    const guild = interaction.guild;
    await guild.members.fetch(); // Ensure full member cache

    const totalMembers = guild.memberCount;
    const bots = guild.members.cache.filter(m => m.user.bot).size;
    const humans = totalMembers - bots;
    const online = guild.members.cache.filter(m => m.presence?.status === 'online').size;
    const roles = guild.roles.cache.size - 1; // exclude @everyone
    const channels = guild.channels.cache.size;
    const textChannels = guild.channels.cache.filter(c => c.type === 0).size;
    const voiceChannels = guild.channels.cache.filter(c => c.type === 2).size;
    const createdAt = `<t:${Math.floor(guild.createdTimestamp / 1000)}:D>`;
    const owner = await guild.fetchOwner();

    const embed = new EmbedBuilder()
      .setColor(0x2d7dd2)
      .setTitle(`📊 ${guild.name} — Server Info`)
      .setThumbnail(guild.iconURL({ dynamic: true, size: 256 }))
      .addFields(
        { name: '👥 Total Members', value: `\`${totalMembers}\``, inline: true },
        { name: '🧑 Humans',        value: `\`${humans}\``,       inline: true },
        { name: '🤖 Bots',          value: `\`${bots}\``,         inline: true },
        { name: '🟢 Online',        value: `\`${online}\``,       inline: true },
        { name: '📜 Roles',         value: `\`${roles}\``,        inline: true },
        { name: '📁 Channels',      value: `\`${channels}\` (💬 ${textChannels} · 🔊 ${voiceChannels})`, inline: true },
        { name: '🎖️ Boost Level',   value: `\`Tier ${guild.premiumTier}\` (${guild.premiumSubscriptionCount} boosts)`, inline: true },
        { name: '👑 Owner',         value: `${owner.user.tag}`, inline: true },
        { name: '📅 Created',       value: createdAt, inline: true },
      )
      .setFooter({ text: 'Champions of the Shattered Realm · Server Info' })
      .setTimestamp();

    await interaction.editReply({ embeds: [embed] });
  },
};

File: Downloads/CSR-System-Final/csr-final/commands/serverinfo.py
import discord
from discord.ext import commands
from discord import app_commands


class ServerInfoCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @app_commands.command(name='serverinfo', description='Server stats — members, bots, channels & more.')
    async def serverinfo(self, interaction: discord.Interaction):
        await interaction.response.defer()
        guild = interaction.guild
        await guild.chunk()  # Ensure full member cache

        total    = guild.member_count
        bots     = sum(1 for m in guild.members if m.bot)
        humans   = total - bots
        online   = sum(1 for m in guild.members if m.status == discord.Status.online)
        roles    = len(guild.roles) - 1  # exclude @everyone
        text_ch  = len([c for c in guild.channels if isinstance(c, discord.TextChannel)])
        voice_ch = len([c for c in guild.channels if isinstance(c, discord.VoiceChannel)])
        total_ch = len(guild.channels)

        owner = guild.owner
        created_ts = int(guild.created_at.timestamp())

        embed = discord.Embed(title=f'📊 {guild.name} — Server Info', color=0x2d7dd2)
        embed.set_thumbnail(url=guild.icon.url if guild.icon else None)
        embed.add_field(name='👥 Total Members', value=f'`{total}`',   inline=True)
        embed.add_field(name='🧑 Humans',        value=f'`{humans}`',  inline=True)
        embed.add_field(name='🤖 Bots',          value=f'`{bots}`',    inline=True)
        embed.add_field(name='🟢 Online',        value=f'`{online}`',  inline=True)
        embed.add_field(name='📜 Roles',         value=f'`{roles}`',   inline=True)
        embed.add_field(name='📁 Channels',      value=f'`{total_ch}` (💬 {text_ch} · 🔊 {voice_ch})', inline=True)
        embed.add_field(name='🎖️ Boost Level',   value=f'`Tier {guild.premium_tier}` ({guild.premium_subscription_count} boosts)', inline=True)
        embed.add_field(name='👑 Owner',         value=str(owner) if owner else 'Unknown', inline=True)
        embed.add_field(name='📅 Created',       value=f'<t:{created_ts}:D>', inline=True)
        embed.set_footer(text='Champions of the Shattered Realm · Server Info')
        embed.timestamp = discord.utils.utcnow()

        await interaction.followup.send(embed=embed)


async def setup(bot):
    await bot.add_cog(ServerInfoCog(bot))
